### PR TITLE
Use a class instead of double to represent the statistics analysis result

### PR DESCRIPTION
**Pull request by carsonwang**
## What changes were proposed in this pull request?
1. Wrap the double value in case class StatsAnalysisResult
2. When analyzing the statistics result, if any result is SKIP_INDEX, we return SKIP_INDEX.
3. Rename StaticsAnalysisResult to StatsAnalysisResult because statics has another meaning, we may either use statistics or stats.

## How was this patch tested?
Existing tests.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/statistics/StatisticsManager.scala [hunk 1]
```diff
@@ -136,34 +136,18 @@ object StatisticsManager {
...
+      conf: Configuration): StatsAnalysisResult = {
+    val fullScanThreshold = conf.getDouble(
+      OapConf.OAP_FULL_SCAN_THRESHOLD.key, OapConf.OAP_FULL_SCAN_THRESHOLD.defaultValue.get)
+    val analysisResults = stats.map(_.analyse(intervalArray))
+
+    if (analysisResults.exists(_ == StatsAnalysisResult.SKIP_INDEX)) {
+      StatsAnalysisResult.SKIP_INDEX
+    } else if (analysisResults.isEmpty ||
+      analysisResults.map(_.coverage).sum / analysisResults.length <= fullScanThreshold) {
+      StatsAnalysisResult.USE_INDEX
```
Comment (carsonwang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/statistics/StatisticsManager.scala]: @adrian-wang , can you help review this part? When analyzing the statistics result, if any result is SKIP_INDEX, I now return SKIP_INDEX.

Comment (adrian-wang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/statistics/StatisticsManager.scala]: This is fine to me

**Comment by LuciferYang** [comment]
I think we needs a fine-grained way to determine which index is more suitable for current sql, perhaps StatsAnalysisResult should include more details and let IndexScanner can be choice by StatsAnalysisResult at executor side , then we can pushdown more than one indexscanner and use StatsAnalysisResult to avoid using the index to be affected by the order of index creation.

**Comment by carsonwang** [comment]
@LuciferYang were you suggesting pushdown more index scanners even if `spark.sql.oap.indexer.max.use.size` is set to 1? How can we determine a better index based on the stats analysis result?

**Comment by LuciferYang** [comment]
@carsonwang yup~ perhaps  the ratio of the current task indexscanner hit recored percentage, but I am not sure. 
In the integration test  there is a typical case as follow:  `select f1, f2, f3 from wise where uid = xxx and cid = xxx`, the uid will hit 20% recoreds and cid will hit 1% records for more tasks, but uid was created first and we can not force use cid to optimization this case because of bigsql is a service. So I think we should give a strategy to compare which index is better even if `spark.sql.oap.indexer.max.use.size`  is set to 1.

**Comment by carsonwang** [comment]
@LuciferYang , thanks for the explanation and it sounds useful. While I heard from @linhong-intel that it will cause the OAP strategies stop working. So let's keep investigating and discussing on what you proposed before we get a clear idea.

**Comment by carsonwang** [comment]
@adrian-wang , is there any other comments regarding the change itself in this PR?

**Comment by LuciferYang** [comment]
@carsonwang ok~ we can discuss it on wechat~

**Comment by carsonwang** [comment]
This is also a bug fix to the statistic issue Baidu met. I am merging this as no more comment.

**Comment by LuciferYang** [comment]
Thx~ @carsonwang

**Comment by carsonwang** [comment]
Also merged to 0.3.

**Pull request merged**